### DISCUSSION TITLE
The cheapest method for change the VBO like D3D11_USAGE_DYNAMIC

### DISCUSSION THREAD

**Discussion by fanbinnet** [google/filament#4620] [Q&A] [answered]
I change the vertexBuffer value in doFrame before renderer.render(view).
 
for (int i = 0; i < mMaxNum; i++) {
            kVertices[i].x = kVertices[i].x;
            kVertices[i].y = kVertices[i].y - 0.01f;
            vertexData.putFloat(kVertices[i].y);
}
vertexBuffer.setBufferAt(mEngine, 0, vertexData);

rm.setGeometryAt(rm.getInstance(renderable),0,
        RenderableManager.PrimitiveType.POINTS, vertexBuffer, indexBuffer, 0, mMaxNum);

The question, I have refer pointSprite.cpp example, If these is the cheapest method for change the VBO value like other particle system set the "VBO Usage = D3D11_USAGE_DYNAMIC".

**Comment by romainguy** [comment, marked as answer]
If you didn't change the number of vertices in the vertex buffer, you don't need to call `setGeometryAt()`.

**Reply by fanbinnet** [reply]
Thanks，remove setGeometryAt can work fine！
